contracts/utility/ExpDecayMath.sol

```diff
@@ -12,12 +12,13 @@ library ExpDecayMath {
      * @dev returns the amount required for a token after a given time period since trading has been enabled
      *
      * the returned value is calculated as `amount / 2 ^ (timeElapsed / halfLife)`
-     * note that because the exponentiation function is limited to an input of up to (and excluding)
-     * 16 / ln 2, the input value to this function is limited by `timeElapsed / halfLife < 16 / ln 2`
+     * note that the input value to this function is limited by `timeElapsed / halfLife <= 128`
      */
     function calcExpDecay(uint256 amount, uint32 timeElapsed, uint32 halfLife) internal pure returns (uint256) {
-        Fraction memory input = Fraction({ n: timeElapsed, d: halfLife });
+        uint256 integerPart = timeElapsed / halfLife;
+        uint256 fractionPart = timeElapsed % halfLife;
+        Fraction memory input = Fraction({ n: fractionPart , d: halfLife });
         Fraction memory output = MathEx.exp2(input);
-        return MathEx.mulDivF(amount, output.d, output.n);
+        return MathEx.mulDivF(amount, output.d, output.n * 2 ** integerPart);
```

Comment:
Are you sure this won't revert in edge case? I.e., if `amount * output.d` or `output.n * 2 ** integerPart` become too huge?

Reply:
I want it to.
Otherwise I would have used `output.n << integerPart`.
BTW, regardless of this change, `mulDivF` itself reverts whenever its final outcome is (still) larger than 256 bits.
The purpose in this PR is to extend the supported input range, not to make it equal to the entire input range.